mlflow/sklearn/utils.py

```diff
@@ -97,21 +127,178 @@ def _truncate_dict(d, max_key_length=None, max_value_length=None):
         should_truncate_key = (not key_is_none) and (len(str(k)) > max_key_length)
         should_truncate_val = (not val_is_none) and (len(str(v)) > max_value_length)
 
-        new_k = str(k)[:max_key_length] if should_truncate_key else k
+        new_k = _truncate_and_ellipsize(k, max_key_length) if should_truncate_key else k
         if should_truncate_key:
-            msg = "Truncated the key `{}`".format(k)
+            # Use the truncated key for warning logs to avoid noisy printing to stdout
+            msg = "Truncated the key `{}`".format(new_k)
             _logger.warning(msg)
 
-        new_v = str(v)[:max_value_length] if should_truncate_val else v
+        new_v = _truncate_and_ellipsize(v, max_value_length) if should_truncate_val else v
         if should_truncate_val:
-            msg = "Truncated the value `{}` (in the key `{}`)".format(v, k)
+            # Use the truncated key and value for warning logs to avoid noisy printing to stdout
+            msg = "Truncated the value of the key `{}`. Truncated value: `{}`".format(new_k, new_v)
             _logger.warning(msg)
 
         truncated[new_k] = new_v
 
     return truncated
 
 
+def _get_meta_estimators_for_autologging():
+    """
+    :return: A list of meta estimator class definitions
+             (e.g., `sklearn.model_selection.GridSearchCV`) that should be included
+             when patching training functions for autologging
+    """
+    from sklearn.model_selection import GridSearchCV, RandomizedSearchCV
+    from sklearn.pipeline import Pipeline
+
+    return [
+        GridSearchCV,
+        RandomizedSearchCV,
+        Pipeline,
+    ]
+
+
+def _is_parameter_search_estimator(estimator):
+    """
+    :return: `True` if the specified scikit-learn estimator is a parameter search estimator,
+             such as `GridSearchCV`. `False` otherwise.
+    """
+    from sklearn.model_selection import GridSearchCV, RandomizedSearchCV
+
+    parameter_search_estimators = [
+        GridSearchCV,
+        RandomizedSearchCV,
+    ]
+
+    return any(
+        [
+            isinstance(estimator, param_search_estimator)
+            for param_search_estimator in parameter_search_estimators
+        ]
+    )
```

Comment:
Not sure if we really need to do this, but this function returns `True` when the estimator is **a subclass** of `GridSearchCV` or `RandomizedSearchCV`.

```python
from sklearn.model_selection import GridSearchCV


class NewGridSearchCV(GridSearchCV):
    def __init__(self):
        pass


ngv = NewGridSearchCV()
print(isinstance(ngv, GridSearchCV))
print(ngv.__class__ == NewGridSearchCV)
print(ngv.__class__ == GridSearchCV)

# output
True
True
False
```

Reply:
Looks like there is no class that inherits from `GridSearchCV` or `RandomizedSearchCV`.

```
import sklearn

from sklearn.model_selection import GridSearchCV, RandomizedSearchCV


for _, class_def in sklearn.utils.all_estimators():
    for func_name in ["fit"]:
        if issubclass(class_def, GridSearchCV) or issubclass(class_def, RandomizedSearchCV):
            print(class_def)

# output
<class 'sklearn.model_selection._search.GridSearchCV'>
<class 'sklearn.model_selection._search.RandomizedSearchCV'>
```

Reply:
makes sense :)